src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -217,4 +292,136 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String queryId() {
         return NAME;
     }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        String field;
+        FieldMapper mapper = parseContext.fieldMapper(fieldName);
+        if (mapper != null) {
+            field = mapper.names().indexName();
+        } else {
+            field = fieldName;
+        }
+
+        Analyzer analyzerObj;
+        if (analyzer == null) {
+            if (mapper != null) {
+                analyzerObj = parseContext.getSearchAnalyzer(mapper);
+            } else {
+                analyzerObj = parseContext.mapperService().searchAnalyzer();
+            }
+        } else {
+            analyzerObj = parseContext.mapperService().analysisService().analyzer(analyzer);
+            if (analyzerObj == null) {
+                throw new IllegalArgumentException("No analyzer found for [" + analyzer + "]");
+            }
+        }
+        
+        Occur highFreqOccur = highFreqOperator.equals(Operator.AND) ? Occur.MUST : Occur.SHOULD;
+        Occur lowFreqOccur = lowFreqOperator.equals(Operator.AND) ? Occur.MUST : Occur.SHOULD;
+
+        ExtendedCommonTermsQuery commonsQuery = new ExtendedCommonTermsQuery(highFreqOccur, lowFreqOccur, cutoffFrequency, disableCoord, mapper);
+        Query query = parseQueryString(commonsQuery, text, field, analyzerObj, lowFreqMinimumShouldMatch, highFreqMinimumShouldMatch);
+
+        query.setBoost(boost);
+        if (queryName != null) {
+            parseContext.addNamedQuery(queryName, query);
+        }
+        return query;
+    }
+    
+    public static final Query parseQueryString(ExtendedCommonTermsQuery query, String queryString, String field, Analyzer analyzer, 
```

Comment:
I use this in CommonTermsQueryBuilderTest#createExpectedQuery. This is related to the duplicate code issue on compounded queries.